frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/advanced-options/LifecycleHookField.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { FormikValues, useField, useFormikContext } from 'formik';
+import { ButtonVariant, Button, Tooltip } from '@patternfly/react-core';
+import { PlusCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import LifecycleHookForm from './LifecycleHookForm';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { getStrategyData } from '../../utils/edit-deployment-utils';
+import { getResourcesType } from '../../../edit-application/edit-application-utils';
+import './LifecycleHookField.scss';
+
+const LifecycleHookField: React.FC<{
+  lifecycleHookName: string;
+  resourceObj: K8sResourceKind;
+  dataAttribute: string;
+  title?: string;
+  subTitle?: string;
+}> = ({ title, subTitle, dataAttribute, lifecycleHookName, resourceObj }) => {
+  const { t } = useTranslation();
+  const {
+    setFieldValue,
+    values: {
+      formData: {
+        name: resName,
+        project: { name: resNamespace },
+        deploymentStrategy,
+      },
+    },
+    initialValues,
+  } = useFormikContext<FormikValues>();
+  const [{ value: lifecycleHookExist }] = useField<boolean>(
+    `formData.deploymentStrategy.${dataAttribute}.${lifecycleHookName}.exists`,
+  );
+  const [showForm, setShowForm] = React.useState(false);
+  const resourceType = getResourcesType(resourceObj);
+
+  const onRemove = () => {
+    const strategy = deploymentStrategy;
+    strategy[dataAttribute][lifecycleHookName] = undefined;
+    const data = getStrategyData(
+      strategy.type,
+      { [dataAttribute]: strategy[dataAttribute] },
+      resName,
+      resNamespace,
+      resourceType,
+    );
+    initialValues.formData.deploymentStrategy[dataAttribute][lifecycleHookName] =
+      data[dataAttribute][lifecycleHookName];
+    initialValues.formData.deploymentStrategy.imageStreamData[lifecycleHookName] =
+      data.imageStreamData[lifecycleHookName];
+    setFieldValue(
+      `formData.deploymentStrategy.${dataAttribute}.${lifecycleHookName}`,
+      data[dataAttribute][lifecycleHookName],
+    );
+    setFieldValue(
+      `formData.deploymentStrategy.imageStreamData.${lifecycleHookName}`,
+      data.imageStreamData[lifecycleHookName],
+    );
+  };
+  return (
+    <div>
```

Comment:
Fragment maybe? There are usages for divs in the case of form/grid usages but otherwise we should avoid unstyled divs 🤔

Reply:
With fragment all form styles were not getting applies properly.

Reply:
With fragment all form styles were not getting applies properly.

Reply:
kk, thanks!